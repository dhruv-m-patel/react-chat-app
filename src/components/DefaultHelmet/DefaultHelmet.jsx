import React from 'react';
import { Helmet } from 'react-helmet';
import PropTypes from 'prop-types';

export default function DefaultHelmet({ title, description }) {
  return (
    <Helmet>
      <title>{title}</title>
      <meta name="description" content={description} />
    </Helmet>
  );
}

DefaultHelmet.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
};

DefaultHelmet.defaultProps = {
  title: 'React App',
  description: 'A boilerplate React app with SSR support',
};
