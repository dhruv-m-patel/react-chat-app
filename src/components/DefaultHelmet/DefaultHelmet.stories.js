import React from 'react';
import { storiesOf } from '@storybook/react';
import DefaultHelmet from './DefaultHelmet';

storiesOf('DefaultHelmet', module)
  .addDecorator(story => <div style={{ margin: '2rem' }}>{story()}</div>)
  .add('with default props', () => (
    <React.Fragment>
      <DefaultHelmet />
      <p>Inspect page markup in HEAD section to know more.</p>
    </React.Fragment>
  ))
  .add('with overrides', () => (
    <React.Fragment>
      <DefaultHelmet
        title="My Custom Title"
        description="This is a custom page description"
      />
      <p>Inspect page markup in HEAD section to know more.</p>
    </React.Fragment>
  ));
