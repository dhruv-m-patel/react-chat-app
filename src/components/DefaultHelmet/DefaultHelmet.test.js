import Tester from '../../../tests/Tester';
import DefaultHelmet from './DefaultHelmet';

const tester = new Tester();

describe('DefaultHelmet', () => {
  test('it should render', () => {
    const snapshot = tester.getSnapshot(DefaultHelmet, {
      title: 'test app',
      'description': 'just testing'
    });
    expect(snapshot).toMatchSnapshot();
  });
});
