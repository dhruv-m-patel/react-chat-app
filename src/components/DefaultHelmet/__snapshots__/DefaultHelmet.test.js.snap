// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultHelmet it should render 1`] = `null`;
